_posts/2019-04-10-aws-dashboard-console-using-arduino.md

```diff
@@ -0,0 +1,265 @@
+               "User-Agent: ArduinoRadiatorConsole\r\n" +
+               "x-api-key: " + api_key + "\r\n" +
+               "Connection: close\r\n\r\n");
+
+  Serial.println("request sent");
+
+  while (client.connected()) {
+    String line = client.readStringUntil('\n');
+
+    if (line == "\r") {
+      Serial.println("headers received");
+      break;
+    }
+  }
+
+  String line = client.readStringUntil('}');
+  line = line + "}";
+
+  client.stop();
+```
+
+The low level library `WiFiClientSecure` is used for the HTTP request. High level HTTP libraries are also available but I wanted to stay in basics and decided to use this low level implementation. The HTTP response is read until a closing parenthesis which is a last character of the HTTP REST response.
+
+After the response is read to the string, it is parsed by a JSON parser. I used the [ArduinoJSON](https://arduinojson.org/) library.
+
+**The physical LED lights and LCD screen content are set inside the `setContent` method:**
+
+```c++
+digitalWrite(PIPE_RUNNING_LED_DEV, LOW);
+digitalWrite(PIPE_FAILED_LED_DEV, LOW);
+digitalWrite(PIPE_RUNNING_LED_TEST, LOW);
+digitalWrite(PIPE_FAILED_LED_TEST, LOW);
+digitalWrite(ALARMS_LED, LOW);
+digitalWrite(PIPE_RUNNING_LED_PROD, LOW);
+digitalWrite(PIPE_FAILED_LED_PROD, LOW);
+
+if (pipelines_running_dev == true) { digitalWrite(PIPE_RUNNING_LED_DEV, HIGH); }
+if (pipelines_failed_dev == true) { digitalWrite(PIPE_FAILED_LED_DEV, HIGH); }
+if (pipelines_running_test == true) { digitalWrite(PIPE_RUNNING_LED_TEST, HIGH); }
+if (pipelines_failed_test == true) { digitalWrite(PIPE_FAILED_LED_TEST, HIGH); }
+if (alarms_raised_prod == true) { digitalWrite(ALARMS_LED, HIGH); }
+if (pipelines_running_prod == true) { digitalWrite(PIPE_RUNNING_LED_PROD, HIGH); }
+if (pipelines_failed_prod == true) { digitalWrite(PIPE_FAILED_LED_PROD, HIGH); } 
+```
+
+First, the LEDs are switch Off to clear the previous state. Then the necessary LEDs are switch On by the simple `if` clauses.
+
+**After the LED’s are fine, write content to the LCD:**
+
+```c++
+  for (int i=0; i < 5; i++) {
+
+    bool all_ok = true;
+  
+    if (pipelines_failed_dev == true) {
+      lcd.setCursor(0,0);
+      lcd.print("DEV Pipe fail:  ");
+      lcd.setCursor(0,1);
+      lcd.print(pipes_failed_0_dev);
+
+      all_ok = false;
+      delay(5000);
+    } 
+    
+    if (alarms_raised_prod == true) {  
+      lcd.setCursor(0,0);
+      lcd.print("PROD Alarms:    ");
+      lcd.setCursor(0,1);
+      lcd.print(alarms_0_prod);
+
+      all_ok = false;
+      delay(5000);
+    }
+
+    ...Other clauses...
+
+    if (all_ok == true) {  
+      lcd.clear();
+      lcd.print("     All OK     ");
+
+      delay(60000);
+
+      break;
+    }
+  }
+```
+
+Every `if` clause contains 5 second delay to keep the certain text on the screen for a moment. The `for` loop is used to “blink” all texts few times on the screen. If there are nothing to show, then print “All OK”.
```

Comment:
5 second delay --> 5-second delay 
If there are nothing to show --> If there is nothing to show